lesson3.md

```diff
@@ -0,0 +1,841 @@
+
+
+---
+
+
+Parentheses have the highest precedence and always win:
+
+```js
+(2 + 2) * 2   // 8
+2 + (2 * 2)   // 6
+```
+
+Always use parentheses when chaining!
+
+It makes your code more clear.
+
+
+---
+
+
+### Quiz
+
+What does the following return?
+
+```js
+(false && false) || true
+```
+
+<br/>
+
+
+```js
+(false && false) || true   // false && false is false
+false || true              // returns true
+```
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### TIP: variables instead of chaining
+
+Which is more understandable?
+
+```js
+42 > 22 && 42 > 35
+```
+
+Or
+
+```js
+let myAge = 42;
+let leftPersonAge = 22;
+let rightPersonAge = 35;
+
+let olderThanLeft = myAge > leftPersonAge;
+let olderThanRight = myAge > rightPersonAge;
+let isOldest = olderThanLeft && olderThanRight;
+```
+
+
+---
+
+
+### Quick Quiz, now in VSCode!
+
+Let's say, everyone here wants to go to the cinema.
+
+How many people in total are going to the cinema?
+<!-- .element: class="fragment" -->
+
+A ticket to watch the movie *The Joker* costs 8€, how much do we have to pay in total?
+<!-- .element: class="fragment" -->
+
+Saskia and I volunteered to pay for the tickets, how much each one pays?
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Quick Quiz (Cont.)
+
+
+I can only pay 100€, is it enough?
+
+
+The cinema will only allow us to enter if we are less than 30 people and if we pay more than 100€, will we get in?
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Recap
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * /` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `temperature !== 25`<br/>`age >= 18` |
+| Logical Operators | &#124;&#124; `&& !` | `a && !b`<br />`x >= 5 && x < 15` |
+| **Combined Operators** | `+= -= *= /= ++ --` | `a *= 2`<br />`count++` |
+
+
+---
+
+
+### Excursus: Combined Operators
+
+This
+
+```js
+let myAge = 42;
+myAge = myAge + 5;
+```
+
+Is the same as this
+
+```js
+let myAge = 42;
+myAge += 5;
+```
+
+
+---
+
+
+### Quiz: What's the price?
+
+```js
+let price = 10;
+
+// there's a 50 percent sale!
+price /= 2;
+
+// price is ??
+```
+
+Result: `5`
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Operator ++ and --
+
+Operator `++` and `--` increase (or decrease) the value of a variable by 1
+
+```js
+let myAge = 42;
+myAge++; // same as myAge = myAge + 1
+```
+
+Combined operators only work with variables!
+
+---
+
+
+<!-- .slide: id="lesson3:if" -->
+# Conditional Statements
+
+
+---
+
+
+### Conditional Statements
+
+Up until now, we wrote code line by line.
+
+But what if I want to run some code only if the user is logged in?
+
+
+---
+
+
+### If statement
+
+```js
+if (condition) {
+  // block of code that
+  // will run ONLY if
+  // condition is true
+}
+```
+
+
+---
+
+
+### If statements
+
+```js
+if (condition) {
+  // your code here
+}
+```
+
+* `condition` must be a boolean
```

Comment:
it must be an expression that evaluates to boolean.

Reply:
expression is such an overloaded term, I like to avoid it when I can.

Reply:
"expression" is the correct term, though :)

since in JavaScript even statements return a value (`undefined`), I guess everything can evaluate to boolean

Reply:
I think it's quite the opposite of overloaded. It has a well defined meaning. 

> 
> BTW. is there anything that you can't evaluate as a boolean?

Point taken. Damn javascript.